src/history.coffee

```diff
@@ -118,13 +133,18 @@ class History
         hasSeenChanges = true
     null
 
-  popChanges: (fromStack, toStack, checkpointIndex) ->
+  popChanges: (fromStack, toStack, checkpointIndex, lookBack) ->
     changes = []
-    snapshot = fromStack[checkpointIndex].snapshot
-    for entry in fromStack.splice(checkpointIndex) by -1
+    snapshotBefore = null
+    snapshotAfter = fromStack[checkpointIndex].snapshot
+    splicedEntries = fromStack.splice(checkpointIndex)
+    for entry in splicedEntries by -1
       toStack?.push(entry)
-      changes.push(entry) unless entry instanceof Checkpoint
-    {changes, snapshot}
+      if entry instanceof Checkpoint
```

Comment:
Again, could this be simplified by assuming the `snapshotBefore` is at the top of the stack?